core/src/main/java/org/apache/accumulo/core/constraints/Constraint.java

```diff
@@ -52,9 +53,20 @@
      * Gets the key extent of the environment.
      *
      * @return key extent
+     * @deprecated since 2.1 use {@link #getTablet()}
      */
+    @Deprecated(since = "2.1.0")
```

Comment:
By deprecating this, it seems the intent is to try to make `org.apache.accumulo.core.constraints` another package in the public API. Rather than add another package to the API, we could instead create a new subpackage in an existing API package (`core.data.constraints`, for example), and start a new interface from scratch there, and deprecate this entire interface. This old interface can extend that one, so any implementation of this deprecated interface would also be an instance of the new interface.

Reply:
> By deprecating this, it seems the intent is to try to make org.apache.accumulo.core.constraints another package in the public API

That was the first option I was proposing, since it would be the least disruptive and least amount of work.

> Rather than add another package to the API, we could instead create a new subpackage in an existing API package (core.data.constraints, for example), and start a new interface from scratch there, and deprecate this entire interface. This old interface can extend that one, so any implementation of this deprecated interface would also be an instance of the new interface.

This is definitely more work and adds code complexity but it would prevent adding another package to the API. Do you see any other changes to the API other than replacing `KeyExtent` with `TabletId`? The rest of the API seemed OK, another reason I didn't think we needed to deprecate the entire interface.

Reply:
> This is definitely more work and adds code complexity but it would prevent adding another package to the API.

This may be worth it. Our previous efforts around defining the public API have tried to keep the API definition simple and succinct. (Like moving out non-API subpackages.) I suspect there are more places like this where we want to move things into public API (iterators/filters, eventually, for example), and don't want to set a precedent for bloating the public API definition.

> Do you see any other changes to the API other than replacing `KeyExtent` with `TabletId`? The rest of the API seemed OK, another reason I didn't think we needed to deprecate the entire interface.

I didn't look closely (apilyzer-maven-plugin should detect any). Even if there aren't, that's fine. It just means that there are fewer differences between the old location and the new location, so it will be easier to migrate. If there were zero differences, then it would only require updating an import statement to migrate to the new interface to avoid the deprecated one.

Reply:
OK I will try creating a new interface with the same name in `org.apache.accumulo.core.data.constraints`. I was able to modify my changes in b14095e so we don't need to use the static method in KeyExtent. With this change and some slight modifications, the Test is now passing. This gives me more confidence in replacing KeyExtent in this change so making a new interface will hopefully just require renaming.